nemo_text_processing/inverse_text_normalization/es/taggers/electronic.py

```diff
@@ -26,22 +36,36 @@ class ElectronicFst(GraphFst):
     and URLS (which get converted to a "protocol" field).
         e.g. c d f uno arroba a b c punto e d u -> tokens { electronic { username: "cdf1" domain: "abc.edu" } }
         e.g. doble ve doble ve doble ve a b c punto e d u -> tokens { electronic { protocol: "www.abc.edu" } }
+    
+    Args:
+        input_case: accepting either "lower_cased" or "cased" input.
     """
 
-    def __init__(self):
+    def __init__(self, input_case: str = INPUT_LOWER_CASED):
         super().__init__(name="electronic", kind="classify")
 
         delete_extra_space = pynutil.delete(" ")
-        alpha_num = (
-            NEMO_ALPHA
-            | pynini.string_file(get_abs_path("data/numbers/digit.tsv"))
-            | pynini.string_file(get_abs_path("data/numbers/zero.tsv"))
+
+        num = pynini.string_file(get_abs_path("data/numbers/digit.tsv")) | pynini.string_file(
+            get_abs_path("data/numbers/zero.tsv")
         )
+        if input_case == INPUT_CASED:
+            num = capitalized_input_graph(num)
+
+        alpha_num = (NEMO_ALPHA | num).optimize()
 
         symbols = pynini.string_file(get_abs_path("data/electronic/symbols.tsv")).invert()
+        if input_case == INPUT_CASED:
+            symbols = capitalized_input_graph(symbols)
 
         accepted_username = alpha_num | symbols
-        process_dot = pynini.cross("punto", ".")
+        dot = pynini.accep("punto")
```

Comment:
Could you explain this just so I'm sure I'm following?

Reply:
graph can convert either 'punto' or (if capitalized) 'Punto' to '.' in these contexts